<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Current Projects</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Lisa K. Johnson</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-binoculars"></span>
     
    Research
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="current_projects.html">Current Projects</a>
    </li>
    <li>
      <a href="past_projects.html">Previous Projects</a>
    </li>
  </ul>
</li>
<li>
  <a href="publications.html">
    <span class="fa fa-cube"></span>
     
    Publications
  </a>
</li>
<li>
  <a href="teaching.html">
    <span class="fa fa-cogs"></span>
     
    Teaching
  </a>
</li>
<li>
  <a href="http://monsterbashseq.wordpress.com/">
    <span class="fa fa-street-view"></span>
     
    Blog
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-gear"></span>
     
    More
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="links.html">Links</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="music.html">Music</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Current Projects</h1>

</div>


<p>Physiologists strive to understand how and why things work in living tissues. One of the many reasons why I chose a physiology graduate program rather than a bioinformatics- or genomics-focused program was that I want to understand the unique adaptations of living tissues in a diversity of organisms. Transcriptional profiling data from high-throughput nucleic acid sequencing technology is becoming a common practice towards revealing the underlying molecular architecture contributing to how living tissues work. While there is an abundance of bioinformatics software available for analyzing RNAseq data, standardization of practices is lacking. To this end, a core objective of my dissertation is to develop a set of reproducible bioinformatics pipeline tools and accompanying tutorials that will provide an open-source framework for analyzing complex multi-species RNAseq datasets. I am currently developing these tools with data pertaining to the following projects:</p>
<div id="re-analysis-of-the-marine-microbial-eukaryotic-sequencing-project" class="section level1">
<h1>Re-analysis of the Marine Microbial Eukaryotic Sequencing Project</h1>
<p>De novo transcriptome assemblies are required prior to analyzing RNAseq data from a species without an existing reference genome or transcriptome. Despite the prevalence of transcriptomic studies, the effects of using different workflows, or ‘pipelines’, on the resulting assemblies are poorly understood. Here, a pipeline was programmatically automated and used to assemble and annotate raw transcriptomic short read data collected by the Marine Microbial Eukaryotic Transcriptome Sequencing Project (MMETSP). Transcriptome assemblies generated through this pipeline were evaluated and compared against assemblies that were previously generated with a pipeline developed by the National Center for Genome Research (NCGR). As the optimum transcriptome is a moving target, improving (or not) with new tools and approaches, automated and programmable pipelines are invaluable for managing the computationally-intensive tasks required for re-processing large sets of samples with revised pipelines. Moreover, automated and programmable pipelines facilitate the comparison of diverse sets of data by ensuring a common evaluation workflow was applied to all samples. Thus, re-assembling existing data with new tools using automated and programmable pipelines may yield more accurate identification of taxon-specific trends across samples in addition to novel and useful products for the community.</p>
</div>
<div id="fundulus-multispecies-osmotic-transcriptome-sequencing-project-fmotsp" class="section level1">
<h1><em>Fundulus</em> Multispecies Osmotic Transcriptome Sequencing Project (FMOTSP)</h1>
<p>Estuaries are characterized by periodic fluxes in salinity. Many species of North American killifish (Fundulus) are estuarine specialists and harbor euryhaline phenotypes. Three clades within Fundulus independently radiated into freshwater environments and have lost their abilities to tolerate high salinity. We use Fundulus as a comparative model system for studying the physiological and genetic mechanisms that diverge between euryhaline and freshwater species. We examined 16 estuarine and freshwater species with representation from each of three clades. Fish from all species were acclimated to either brackish or fresh water then exposed to an acute brackish water challenge. Transcriptome data from gill epithelium tissue were collected. To enable multi-species comparisons, reference transcriptome assemblies were generated de novo for each species then used to analyze transcriptional responses to salinity change by clade and physiology.</p>
</div>
<div id="freshwater-fundulus-killifish-genome-sequencing" class="section level1">
<h1>Freshwater <em>Fundulus</em> killifish genome sequencing</h1>
<p>Using Oxford Nanopore Technologies (ONT) long reads to sequence freshwater killifish representing each indepdent freshwater clade (<em>F. olivaceus</em>, <em>F. nottii</em>, and <em>F. catanatus</em>), genome assemblies and annotations are underway, enabling us to scan for enrichment patterns in regulatory regions and also to determine whether or not patterns in transcriptional expression across treatment groups are due to genomic modifications.</p>
<div class="figure">
<img src="img/killifishDNA.jpg" />

</div>
<div class="figure">
<img src="img/code.png" />

</div>
</div>

<html>

<head>
<title>footer</title>
</head>

<body>

<p>Copyright &copy; 2018 Lisa K. Johnson</p>

</body>
</html>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
